jdaviz/core/tools.py

```diff
@@ -338,6 +339,39 @@ def is_visible(self):
         return len([m for m in self.viewer.figure.marks if isinstance(m, SpectralLine)]) > 0
 
 
+@viewer_tool
+class StretchBounds(CheckableTool):
+    icon = os.path.join(ICON_DIR, 'line_select.svg')
+    tool_id = 'jdaviz:stretch_bounds'
+    action_text = 'Set Stretch VMin and VMax'
+    tool_tip = 'Set closest stretch bound (VMin/VMax) with click or click+drag'
+
+    def __init__(self, viewer, **kwargs):
+        self._time_last = 0
+        super().__init__(viewer, **kwargs)
+
+    def activate(self):
+        self.viewer.add_event_callback(self.on_mouse_event,
+                                       events=['dragmove', 'click'])
+
+    def deactivate(self):
+        self.viewer.remove_event_callback(self.on_mouse_event)
+
+    def on_mouse_event(self, data):
+        if (time.time() - self._time_last) <= 0.2:
+            # throttle to 200ms
+            return
```

Comment:
do we need to throttle even?  If so, this needs to re-set `self._time_last`

Reply:
I'm surprised it was even working and not getting immediately throttled when I was testing it, given that I accidentally deleted the reset.

Reply:
I added the time update, and decreased the throttle from 0.2 seconds to 0.05 seconds. 0.2 was slow enough to feel a little unresponsive, 0.05 seems like a good compromise between feeling fast and not throttling at all.